itests/transfer_tx_api_test.go

```diff
@@ -0,0 +1,156 @@
+
+				utl.WavesDiffBalanceCheck(suite.T(), td.Expected.WavesDiffBalanceRecipient,
+					diffBalancesRecipient.DiffBalanceWaves.BalanceInWavesGo,
+					diffBalancesRecipient.DiffBalanceWaves.BalanceInWavesScala, name, "Version: ", v)
+
+				utl.AssetBalanceCheck(suite.T(), td.Expected.AssetDiffBalance,
+					diffBalancesRecipient.DiffBalanceAsset.BalanceInAssetGo,
+					diffBalancesRecipient.DiffBalanceAsset.BalanceInAssetScala, name, "Version: ", v)
+			})
+		}
+	}
+}
+
+func (suite *TransferTxApiSuite) Test_TransferTxApiMaxAmountAndFeePositive() {
+	versions := testdata.GetVersions()
+	waitForTx := true
+	for _, v := range versions {
+		n := transfer_utilities.GetNewAccountWithFunds(&suite.BaseSuite, v, utl.TestChainID,
+			utl.DefaultAccountForLoanFunds, 10000000000)
+		itxID := issue_utilities.IssueAssetAmount(&suite.BaseSuite, v, utl.TestChainID,
+			utl.DefaultSenderNotMiner, utl.MaxAmount)
+		transfer_utilities.TransferAssetAmount(&suite.BaseSuite, v, utl.TestChainID, itxID, utl.DefaultSenderNotMiner, n)
+		tdmatrix := testdata.GetTransferMaxAmountPositive(&suite.BaseSuite, itxID, n)
+
+		for name, td := range tdmatrix {
+			suite.T().Run(name, func(t *testing.T) {
+				tx, diffBalancesSender, diffBalancesRecipient := transfer_utilities.BroadcastTransferTxAndGetBalances(
+					&suite.BaseSuite, td, v, waitForTx)
+
+				utl.StatusCodesCheck(suite.T(), http.StatusOK, http.StatusOK, tx, name, "version", v)
+				utl.TxInfoCheck(suite.T(), tx.WtErr.ErrWtGo, tx.WtErr.ErrWtScala, name,
+					"Transfer: "+tx.TxID.String(), "Version: ", v)
+
+				utl.WavesDiffBalanceCheck(suite.T(), td.Expected.WavesDiffBalanceSender,
+					diffBalancesSender.DiffBalanceWaves.BalanceInWavesGo,
+					diffBalancesSender.DiffBalanceWaves.BalanceInWavesScala, name, "Version: ", v)
+
+				utl.AssetBalanceCheck(suite.T(), td.Expected.AssetDiffBalance,
+					diffBalancesSender.DiffBalanceAsset.BalanceInAssetGo,
+					diffBalancesSender.DiffBalanceAsset.BalanceInAssetScala, name, "Version: ", v)
+
+				utl.WavesDiffBalanceCheck(suite.T(), td.Expected.WavesDiffBalanceRecipient,
+					diffBalancesRecipient.DiffBalanceWaves.BalanceInWavesGo,
+					diffBalancesRecipient.DiffBalanceWaves.BalanceInWavesScala, name, "Version: ", v)
+
+				utl.AssetBalanceCheck(suite.T(), td.Expected.AssetDiffBalance,
+					diffBalancesRecipient.DiffBalanceAsset.BalanceInAssetGo,
+					diffBalancesRecipient.DiffBalanceAsset.BalanceInAssetScala, name, "Version: ", v)
+			})
+		}
+	}
+}
+
+func (suite *TransferTxApiSuite) Test_TransferTxApiNegative() {
+	versions := testdata.GetVersions()
+	waitForTx := true
+	for _, v := range versions {
+		reissuable := testdata.GetCommonIssueData(&suite.BaseSuite).Reissuable
+		itx := issue_utilities.IssueSendWithTestData(&suite.BaseSuite, reissuable, v, waitForTx)
+		tdmatrix := testdata.GetTransferNegativeData(&suite.BaseSuite, itx.TxID)
+
+		if v > 2 {
+			maps.Copy(tdmatrix, testdata.GetTransferChainIDNegativeData(&suite.BaseSuite, itx.TxID))
+		}
```

Comment:
I think it looks better, but I still don't like it.
```go
		if v >= proto.ProtobufTransactionsVersions[proto.TransferTransaction] {
			maps.Copy(tdmatrix, testdata.GetTransferChainIDNegativeData(&suite.BaseSuite, itx.TxID))
		}
```
Ideally you need to make `proto.ProtobufTransactionsVersions` map visibility package private and create a function which you could use for getting protobuf tx version.

Reply:
It will be implemented in new PR.